### PR TITLE
WIP: Add an option not to fix entry points

### PR DESCRIPTION
**Pull request by hmaarrfk**

### CODE REVIEW

**Comment by chrisjbillington** [comment]
Great! I will happily include this. What you're using it for is a bit non-kosher, but so is fixing entry points in the first place :p.

This seems to be a mistake:

```python
            if no_fix_entry_points:
                fix_entry_points(dist)
```

Which should be 

```python
            if not no_fix_entry_points:
                fix_entry_points(dist)
```

I've seen the advice to avoid booleans with "no" or "not" in their names since it makes this mistake easier to make. But `fix_entry_points` is already defined as a function, so a boolean named `fix_entry_points` is out of the question and alternate names for the boolean are probably awkward too. So let's go with the admittedly awkward `if not no_fix_entry_points`.

**Comment by hmaarrfk** [comment]
I wanted to leave the default as fixing the entry point. Thanks for finding the bug

**Comment by hmaarrfk** [comment]
i never had a chance to fix this due to my windows machine being shut down

**Comment by chrisjbillington** [comment]
Yes, I agree with the default remaining as-is, and the command line argument being "--no-fix-entry-points". I started writing a comment about how you could do:

```python
parser.add_argument(
        '--no-fix-entry-points',
        dest='fix_entry_points',
        default=True,
        action='store_false',
        help="If set, entry points will not be fixed for conda-environments "
        "during the install action."
)
```
To keep the command line argument as-is but have the variable in code called `fix_entry_points`, but then noticed the name collision with the function `fix_entry_points` and thought it wasn't worth it.

I'll make the fix and merge it, don't worry about it!

File: desktop_app/shell.py [hunk 1]
```diff
@@ -257,7 +257,8 @@ def install(module_name, path=None, verbose=False):
             # does not support gui_scripts, so those entry_points may have been
             # installed as console_scripts:
             dist = get_distribution_of_module(module_name)
-            fix_entry_points(dist)
+            if no_fix_entry_points:
```
Comment (hmaarrfk) [inline desktop_app/shell.py]:
```suggestion
            if not no_fix_entry_points:
```

**Comment by chrisjbillington** [comment]
Thanks @hmaarrfk! Released as v0.2.9

**Comment by chrisjbillington** [comment]
Build automation is failing due to [a bug](https://github.com/conda-incubator/setup-miniconda/issues/166) in a github action I use, I'll re-run the release workflow once they fix it.

**Comment by hmaarrfk** [comment]
would you be able to retrigger the failed builds on the tag?

**Comment by chrisjbillington** [comment]
I tried, but it just failed again since the bug in the miniconda GitHub action hasn't been fixed yet.

**Comment by chrisjbillington** [comment]
If they don't merge the fix soon I'll update the workflow to work around it

**Comment by hmaarrfk** [comment]
ah i see.

Yeah, I was wondering if it would be possible to use the ctypes API to make this package noarch.

I find it silly that a build script step is forced to use a compiled version of the code instead of a pure python package.

**Comment by hmaarrfk** [comment]
I'm pretty sure you and I are the only users fo this for now.

I'm going to go ahead and patch it up on conda-forge

https://github.com/conda-forge/desktop-app-feedstock/pull/13

**Pull request merged**